source/fundamentals/crud/read-operations/change-streams.txt

```diff
@@ -0,0 +1,266 @@
+sections:
+
+- :ref:`<java-change-stream-open>`
+- :ref:`<java-change-stream-aggregation>`
+- :ref:`<java-change-stream-configure-pre-post>`
+
+.. _java-change-stream-open:
+
+Open a Change Stream
+--------------------
+
+You can open a change stream to subscribe to specific types of data changes
+and produce change events in your application.
+
+To open a change stream, call the ``watch()`` method on an instance of a
+``MongoCollection``, ``MongoDatabase``, or ``MongoClient``.
+
+.. important::
+
+   Standalone MongoDB deployments don't support change streams because
+   the feature requires a replica set oplog. To learn more about the oplog,
+   see the :ref:`<replica-set-oplog>` server manual page.
+
+The object on which you call the ``watch()`` method on determines the scope of
+events that the change stream listens for.
+
+If you call ``watch()`` on a ``MongoCollection``, the change stream monitors
+a collection.
+
+If you call ``watch()`` on a ``MongoDatabase``, the change stream monitors all
+collections in that database.
+
+If you call ``watch()`` on a ``MongoClient``, the change stream monitors all
+changes in the connected MongoDB deployment.
+
+Example
+~~~~~~~
+
+The following code example shows how to open a change stream and print
+change stream events whenever the data in the collection changes:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/change-streams/ChangeStreams.java
+   :language: java
+   :dedent:
+   :start-after: begin openChangeStreamExample
+   :end-before: end openChangeStreamExample
+
+An insert operation on the collection should produce output similar to the
+following text:
+
+.. code-block::
+   :copyable: false
+
+   Received a change: ChangeStreamDocument{
+      operationType='insert',
+      resumeToken={"_data": "825EC..."},
+      namespace=myDb.myChangeStreamCollection,
+      ...
+   }
+
+For a runnable example, see the :ref:`<java-usage-watch>` usage example page.
+
+To learn more about the ``watch()`` method, see the following API
+documentation:
+
+- `MongoCollection.watch() <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#watch()>`__
+- `MongoDatabase.watch() <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoDatabase.html#watch()>`__
+- `MongoClient.watch() <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoClient.html#watch()>`__
+
+.. _java-change-stream-aggregation:
+
+Apply Aggregation Operators to your Change Stream
+-------------------------------------------------
+
+You can pass an aggregation pipeline as a parameter to the ``watch()`` method
+to specify which change events the change stream receives.
+
+To learn which aggregation operators your {+mdb-server+} version supports, see
+:ref:`<change-stream-modify-output>`.
+
+
+Example
+~~~~~~~
+
+The following code example shows how you can apply an aggregation pipeline to
+your change stream receive change events for insert and update operators
+only:
```

Comment:
This sentence was a bit hard to read and left me confused at what the example was showing. I think be easier to understand as something like this (although I'm not sure if modify is the best word to use in the suggestion):

```suggestion
The following code example shows how you can apply an aggregation pipeline to modify
your change stream to only receive change events for insert and update operators:
```